js/F8Navigator.js

```diff
@@ -74,95 +76,92 @@ var F8Navigator = React.createClass({
         return true;
       }
     }
-
-    const {navigator} = this.refs;
-    if (navigator && navigator.getCurrentRoutes().length > 1) {
-      navigator.pop();
-      return true;
-    }
-
-    if (this.props.tab !== 'schedule') {
-      this.props.dispatch(switchTab('schedule'));
-      return true;
-    }
-    return false;
+    this.props.dispatch(back());
+    // Ideally we could return false in the case that the reducer does not change
+    // the state, but we have no way to detect that from this function. Instead of
+    // returning false from here, we can exit the app on android by calling
+    // BackAndroid.exitApp from the navigation reducer.
+    return true;
   },
 
   render: function() {
     return (
-      <Navigator
-        ref="navigator"
+      <NavigationExperimental.AnimatedView
         style={styles.container}
-        configureScene={(route) => {
-          if (Platform.OS === 'android') {
-            return Navigator.SceneConfigs.FloatFromBottomAndroid;
-          }
-          // TODO: Proper scene support
-          if (route.shareSettings || route.friend) {
-            return Navigator.SceneConfigs.FloatFromRight;
-          } else {
-            return Navigator.SceneConfigs.FloatFromBottom;
+        renderScene={this.renderScene}
+        onNavigate={(action) => {
+          if (action.type === 'back') {
+            this.props.dispatch(back());
           }
         }}
-        initialRoute={{}}
-        renderScene={this.renderScene}
+        navigationState={{
+          key: 'F8NavigatorState',
+          index: this.props.openModals.length - 1,
+          children: this.props.openModals,
+        }}
       />
     );
   },
 
-  renderScene: function(route, navigator) {
-    if (route.allSessions) {
-      return (
-        <SessionsCarousel
-          {...route}
-          navigator={navigator}
-        />
-      );
+  renderScene: function(props) {
+    const sceneState = props.scene.navigationState;
+    let isVertical = true;
+    if (sceneState.type === 'ShareSettings') {
+      isVertical = false;
     }
-    if (route.session) {
+    return (
+      <F8NavigationCard
+        isVertical={isVertical}
+        renderScene={this.renderInnerScene}
+        {...props}>
+        {this.renderInnerScene(props)}
```

Comment:
It looks like the {this.renderInnerScene(props)} doesn't get used anywhere.  F8NavigationCard uses this.props.renderScene to render the scene, and doesn't use this.props.children anywhere